src/internal/connector/graph_connector_onedrive_test.go

```diff
@@ -439,185 +531,220 @@ func (suite *GraphConnectorOneDriveIntegrationSuite) TestPermissionsRestoreAndBa
-					items:    folderBReadPerms,
 				},
 				{
-					pathElements: []string{
-						"drives",
-						driveID,
-						"root:",
-						"b",
+					pathElements: folderPath,
+					files: []itemData{
+						{
+							name: fileName,
+							data: fileEData,
+							perms: permData{
+								user:  suite.secondaryUser,
+								roles: writePerm,
+							},
+						},
+					},
+					perms: permData{
+						user:  suite.secondaryUser,
+						roles: readPerm,
 					},
-					category: path.FilesCategory,
-					items:    fileAWritePerms,
-					auxItems: []itemInfo{fileWritePerms},
 				},
 			},
 		},
-
 		{
-			name:     "FolderAndNoChildPermissionsRestore",
-			service:  path.OneDriveService,
-			resource: Users,
-			collections: []colInfo{
+			name:         "FolderAndNoChildPermissionsRestore",
+			startVersion: startVersion,
+			cols: []onedriveColInfo{
 				{
-					pathElements: []string{
-						"drives",
-						driveID,
-						"root:",
+					pathElements: rootPath,
+					folders: []itemData{
+						{
+							name: folderBName,
+							perms: permData{
+								user:  suite.secondaryUser,
+								roles: readPerm,
+							},
+						},
 					},
-					category: path.FilesCategory,
-					items:    folderBReadPerms,
 				},
 				{
-					pathElements: []string{
-						"drives",
-						driveID,
-						"root:",
-						"b",
+					pathElements: folderPath,
+					files: []itemData{
+						{
+							name: fileName,
+							data: fileEData,
+						},
+					},
+					perms: permData{
+						user:  suite.secondaryUser,
+						roles: readPerm,
 					},
-					category: path.FilesCategory,
-					items:    fileEEmptyPerms,
-					auxItems: []itemInfo{fileEmptyPerms},
 				},
 			},
 		},
 	}
 
 	for _, test := range table {
-		suite.T().Run(test.name, func(t *testing.T) {
-			runRestoreBackupTest(t,
-				suite.acct,
-				test,
-				suite.connector.tenant,
-				[]string{suite.user},
-				control.Options{
-					RestorePermissions: true,
-					ToggleFeatures:     control.Toggles{EnablePermissionsBackup: true},
-				},
-			)
-		})
+		expected := testDataForInfo(suite.T(), test.cols, backup.Version)
+
+		for version := test.startVersion; version <= backup.Version; version++ {
+			suite.T().Run(fmt.Sprintf("%s_Version%d", test.name, version), func(t *testing.T) {
+				input := testDataForInfo(t, test.cols, version)
+
+				testData := restoreBackupInfoMultiVersion{
+					service:             path.OneDriveService,
+					resource:            Users,
+					backupVersion:       version,
+					countMeta:           version == 0,
+					collectionsPrevious: input,
+					collectionsLatest:   expected,
+				}
+
+				runRestoreBackupTestVersions(
+					t,
+					suite.acct,
+					testData,
+					suite.connector.tenant,
+					[]string{suite.user},
+					control.Options{
+						RestorePermissions: true,
+						ToggleFeatures:     control.Toggles{EnablePermissionsBackup: true},
+					},
+				)
+			})
+		}
 	}
 }
 
+// TODO(ashmrtn): What this test is supposed to do needs investigated. It
```

Comment:
Woops, this one is on me. This was supposed to test the situation where user has not provided `--restore-permissions`, but as you mentioned, it does not look like that is what it is doing.

Reply:
I will pick up this one in another PR and fix it  in case you don't wanna get to it in this one.

Reply:
fixing in another PR sounds great. If you don't mind picking it up I would appreciate it. If you don't want to though I can take a stab at it